### ISSUE TITLE
IE throws "Command line server for the IE driver has stopped working" error on Click

### ISSUE THREAD

**Issue by BrandonDudek** [SeleniumHQ/selenium#6085]
## Meta -
**OS:**  Windows 7
**Selenium Version:**  3.13.0
**Browser:**  Internet Explorer 11
**Browser Version:**  11.0.9600.19002
**Driver:** IE Driver Server x32 v3.13.0
**Java:** 1.8.0_152

## Expected Behavior 
Click should be error free.

## Actual Behavior
"Command line server for the IE driver has stopped working" error.
![image](https://user-images.githubusercontent.com/19555354/41986740-00db3382-7a05-11e8-928a-580d43938459.png)

## Steps to reproduce
Grab any element and perform a click on it.
**Demo Project:** https://github.com/BrandonDudek/selenium-bug-ie-click

## Last Working
This was working in **Selenium 3.12** + **IE Driver Server x32 v3.12.0**.

## Related Bugs
#2580

## Logs
> T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(232) Entering Server::ProcessRequest
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(296) Entering Server::ReadRequestBody
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(241) Process request with: URI: /session/c4f2ac84-7d24-49aa-8b64-1952f2a05337/element/1e5fa60f-dcd8-498e-b4a4-14a41d98dba8/click HTTP verb: POST
body: {
  "id": "1e5fa60f-dcd8-498e-b4a4-14a41d98dba8"
}
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(330) Entering Server::DispatchCommand
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(654) Entering Server::LookupCommand
D 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(339) Command: POST /session/c4f2ac84-7d24-49aa-8b64-1952f2a05337/element/1e5fa60f-dcd8-498e-b4a4-14a41d98dba8/click {
  "id": "1e5fa60f-dcd8-498e-b4a4-14a41d98dba8"
}
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\server.cc(449) Entering Server::LookupSession
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\iedriver\iesession.cpp(191) Entering IESession::ExecuteCommand
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(130) Entering IECommandExecutor::OnSetCommand
T 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\command.cc(31) Entering Command::Deserialize
D 2018-06-27 13:12:54:996 c:\projects\webdriver\cpp\webdriver-server\command.cc(36) Raw JSON command: { "name" : "click", "locator" : { "sessionid" : "c4f2ac84-7d24-49aa-8b64-1952f2a05337", "id" : "1e5fa60f-dcd8-498e-b4a4-14a41d98dba8" }, "parameters" : {
  "id": "1e5fa60f-dcd8-498e-b4a4-14a41d98dba8"
} }
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(141) Entering IECommandExecutor::OnExecCommand
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iesession.cpp(212) Beginning wait for response length to be not zero
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(613) Entering IECommandExecutor::DispatchCommand
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(793) Entering IECommandExecutor::GetCurrentBrowser
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(799) Entering IECommandExecutor::GetManagedBrowser
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(739) Entering IECommandExecutor::IsAlertActive
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\browser.cpp(714) Entering Browser::GetActiveDialogWindowHandle
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\browser.cpp(218) Entering Browser::GetContentWindowHandle
D 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(753) No alert handle is found
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(793) Entering IECommandExecutor::GetCurrentBrowser
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(799) Entering IECommandExecutor::GetManagedBrowser
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandhandler.cpp(45) Entering IECommandHandler::GetElement
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(887) Entering IECommandExecutor::GetManagedElement
T 2018-06-27 13:12:54:997 c:\projects\webdriver\cpp\iedriver\elementrepository.cpp(34) Entering ElementRepository::GetManagedElement
T 2018-06-27 13:12:55:005 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(793) Entering IECommandExecutor::GetCurrentBrowser
T 2018-06-27 13:12:55:005 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(799) Entering IECommandExecutor::GetManagedBrowser
T 2018-06-27 13:12:55:005 c:\projects\webdriver\cpp\iedriver\browser.cpp(138) Entering Browser::GetDocument
I 2018-06-27 13:12:55:005 c:\projects\webdriver\cpp\iedriver\browser.cpp(142) No child frame focus. Focus is on top-level frame
T 2018-06-27 13:12:55:010 c:\projects\webdriver\cpp\iedriver\browser.cpp(627) Entering Browser::GetDocumentFromWindow
T 2018-06-27 13:12:55:016 c:\projects\webdriver\cpp\iedriver\element.cpp(85) Entering Element::ConvertToJson
T 2018-06-27 13:12:55:016 c:\projects\webdriver\cpp\iedriver\element.cpp(547) Entering Element::GetLocationOnceScrolledIntoView
T 2018-06-27 13:12:55:016 c:\projects\webdriver\cpp\iedriver\element.cpp(679) Entering Element::GetLocation
T 2018-06-27 13:12:55:017 c:\projects\webdriver\cpp\iedriver\element.cpp(815) Entering Element::IsInline
D 2018-06-27 13:12:55:017 c:\projects\webdriver\cpp\iedriver\element.cpp(725) Element is a block element, using IHTMLElement2::getBoundingClientRect
T 2018-06-27 13:12:55:025 c:\projects\webdriver\cpp\iedriver\element.cpp(836) Entering Element::RectHasNonZeroDimensions
T 2018-06-27 13:12:55:027 c:\projects\webdriver\cpp\iedriver\element.cpp(852) Entering Element::GetFrameDetails
T 2018-06-27 13:12:55:027 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument
D 2018-06-27 13:12:55:035 c:\projects\webdriver\cpp\iedriver\element.cpp(802) Element is not in a frame
T 2018-06-27 13:12:55:035 c:\projects\webdriver\cpp\iedriver\element.cpp(1093) Entering Element::CalculateClickPoint
T 2018-06-27 13:12:55:035 c:\projects\webdriver\cpp\iedriver\element.cpp(1030) Entering Element::GetClickableViewPortLocation
T 2018-06-27 13:12:55:035 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument
T 2018-06-27 13:12:55:036 c:\projects\webdriver\cpp\iedriver\documenthost.cpp(304) Entering DocumentHost::GetDocumentMode
T 2018-06-27 13:12:55:049 c:\projects\webdriver\cpp\iedriver\element.cpp(1119) Entering Element::IsLocationInViewPort
T 2018-06-27 13:12:55:049 c:\projects\webdriver\cpp\iedriver\element.cpp(1030) Entering Element::GetClickableViewPortLocation
T 2018-06-27 13:12:55:049 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument
T 2018-06-27 13:12:55:050 c:\projects\webdriver\cpp\iedriver\documenthost.cpp(304) Entering DocumentHost::GetDocumentMode
T 2018-06-27 13:12:55:056 c:\projects\webdriver\cpp\iedriver\element.cpp(614) Entering Element::IsHiddenByOverflow
T 2018-06-27 13:12:55:056 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument
T 2018-06-27 13:12:55:057 c:\projects\webdriver\cpp\iedriver\script.cpp(60) Entering Script::Initialize
T 2018-06-27 13:12:55:057 c:\projects\webdriver\cpp\iedriver\script.cpp(110) Entering Script::AddArgument(IHTMLElement*)
T 2018-06-27 13:12:55:057 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-06-27 13:12:55:057 c:\projects\webdriver\cpp\iedriver\script.cpp(180) Entering Script::Execute
T 2018-06-27 13:12:55:057 c:\projects\webdriver\cpp\iedriver\script.cpp(555) Entering Script::CreateAnonymousFunction
D 2018-06-27 13:12:55:100 c:\projects\webdriver\cpp\iedriver\element.cpp(593) (x, y, w, h): 263, 325, 461, 34
T 2018-06-27 13:12:55:100 c:\projects\webdriver\cpp\iedriver\element.cpp(94) Entering Element::IsDisplayed
T 2018-06-27 13:12:55:100 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument
T 2018-06-27 13:12:55:101 c:\projects\webdriver\cpp\iedriver\script.cpp(60) Entering Script::Initialize
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(110) Entering Script::AddArgument(IHTMLElement*)
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(99) Entering Script::AddArgument(bool)
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(180) Entering Script::Execute
T 2018-06-27 13:12:55:102 c:\projects\webdriver\cpp\iedriver\script.cpp(555) Entering Script::CreateAnonymousFunction
T 2018-06-27 13:12:55:119 c:\projects\webdriver\cpp\iedriver\element.cpp(1144) Entering Element::GetContainingDocument

## Attempted Workarounds
Here are things that I have tried, with no success.

### enableNativeEvents
I tried removing the command: `options.enablePersistentHovering();`
And adding the commands:
`options.requireWindowFocus();
options.enableNativeEvents();`

### Not destructivelyEnsureCleanSession
I tried removing the command: `options.destructivelyEnsureCleanSession();`

### Not introduceFlakinessByIgnoringSecurityDomains
I tried removing the command: `options.introduceFlakinessByIgnoringSecurityDomains();`
With "protected mode" enabled.

**Comment by walliski** [comment]
I am also seeing this after updating C# framework and IE 32 bit driver from 3.12 to 3.13.

```
error: The HTTP request to the remote WebDriver server for URL http://localhost:12701/session/4785db57-ed79-40c2-a29c-2740a8f065c8/element/8e29f557-3d36-42c3-98ce-7c3dc97183d3/click timed out after 60 seconds.
```

and

```
System.InvalidOperationException: MaximumRetryTimes was exceeded ---> OpenQA.Selenium.WebDriverException: The HTTP request to the remote WebDriver server for URL http://localhost:57536/session/5df45a42-965c-46e5-a26f-52ff7dbbf07a/url timed out after 60 seconds. ---> System.Net.WebException: The operation has timed out
```

Suppose it is the same thing.

**Comment by rokim** [comment]
I assume you're running Win7 x64, so try to use the x64 IEDriver. That worked for me with the exact same setup.

**Comment by sergmal** [comment]
Having also same error after switching 3.12->3.13
Downgraded back to 3.12 atm

**Comment by BrandonDudek** [comment]
@rokim, Selenium / IE Driver running on Win 7 x64, _**shuold**_ work for both IE 32 & 64 bit.

Also, IE 64 is not an option for me right now, because of Issue #5116 (IE x64 slow typing).

**Comment by rokim** [comment]
@BrandonDudek I totally agree with you "should work for both IE 32 & 64 bit". 
The slow typing bug hit me too... Try to enable `requireWindowFocus`.

```java
InternetExplorerOptions options = new InternetExplorerOptions().requireWindowFocus();
WebDriver driver = new InternetExplorerDriver(options);
```

This is my full setup for WebDriver with IEDriverServer ver. 3.13 x64:
```java
    File ieTemp = new File("C:\\temp\\");
    Builder builder = new InternetExplorerDriverService.Builder()
        .withExtractPath(ieTemp);
    InternetExplorerDriverService service = builder.build();
    
    InternetExplorerOptions options = new InternetExplorerOptions()
        .introduceFlakinessByIgnoringSecurityDomains()
        .requireWindowFocus()
        .destructivelyEnsureCleanSession();
    
    WebDriver driver = new InternetExplorerDriver(service, options);
```

**Comment by jimevans** [comment]
@BrandonDudek Using a port of the code you referenced, I am unable to reproduce the issue you’re describing. Bear in mind, however, that I do not have a Windows 7 environment easily at hand, and that could very well be the deciding factor. It would be interesting to see if a Windows 10 environment reproduces the issue for you. I’m also on holiday at the moment, so it’s unlikely I’ll get the chance to investigate further for some time to come.

**Comment by jimevans** [comment]
Inadvertently closed. Sorry.

**Comment by RoakyWood** [comment]
Same issue, as of 3.13

IE 11.0.9600.19035 Win 7 SP1 requires Actions/MoveTo in order to click.

Chrome Version 67.0.3396.99 (Official Build) (64-bit) Works as before

Best Regards RW

(Many of us are at the mercy of clients with regards to old OS/Browser combos until they upgrade)

**Comment by jimevans** [comment]
@RoakyWood Since 3.13 is the version this issue was reported against, I'm not sure what you're implying with your comment. Once again, I completely acknowledge that it may be a Windows 7 issue, but I cannot be certain at present. It could also be related to the specific site being automated. Without being able to distinguish between the two, it's impossible to tell. This is why I said, "It would be interesting to see if a Windows 10 environment reproduces the issue for you." I realize people may be constrained to what OS they are running; I need someone who can operate in both environments to validate whether it reproduces in both, or only in Windows 7, since I'm unable to do so at present, and am unlikely to be able to get a chance to do so anytime soon.

**Comment by RoakyWood** [comment]
Tested Against 
OS: 
Win 10 Enterprise 

Browsers:
Chrome, 
IE, 
Edge

I ran these two different methods in sequence

                **//Actions method: Good on each OS-Browser combo above**
                Console.WriteLine("Select a Widget");
                Actions getWidgetarrow = new Actions(alldriver);               getprevarrow.MoveToElement(alldriver.FindElement(By.XPath("somepath"))).Click().Build().Perform();
                Thread.Sleep(3000);
                getWidgetarrow.Click().Build().Perform();
                Thread.Sleep(3000);

                **//IWebElement Method: Errors on on IE 11
                Console.WriteLine("Select a Widget");
                IWebElement getWidgetarrow2 = alldriver.FindElement(By.XPath("somepath"));
                Thread.Sleep(3000);
                getWidgetarrow2.Click();
                Thread.Sleep(3000);

Chrome  PASS
Edge       PASS
IE 11       ERROR

_OpenQA.Selenium.WebDriverException: 
A exception with a null response was thrown sending an HTTP request to the remote WebDriver server for URL http://localhost:blah/blah/element/blah/click

The status of the exception was ReceiveFailure, and the message was: 
The underlying connection was closed: An unexpected error occurred on a receive. ---> 

System.Net.WebException: 
The underlying connection was closed: An unexpected error occurred on a receive. ---> 

System.IO.IOException: 
Unable to read data from the transport connection: 

An existing connection was forcibly closed by the remote host. ---> 

System.Net.Sockets.SocketException: 
An existing connection was forcibly closed by the remote host_

P.S.
There was no implication intended towards you at all. It was meant to be a rather innocuous statement to my peers about feeling their pain about preferred technologies vs. client demands.

Peace, RW

**Comment by barancev** [comment]
@RoakyWood Did you see a system error dialogue too? Or just a client-side exception?

**Comment by barancev** [comment]
I should note that I couldn't reproduce the issue using the example project provided in the original report, I ran 100 iterations and got no failure. Win7, IE11, Selenium + IEDriverServer v.3.13.0

**Comment by RoakyWood** [comment]
IWebElement Click method worked prior to 3.13 for years without error. I upgrade to 3.13 it stops working, throwing that ‘Dispose’ error. Several others have reported this defect. 3.13 also seems to have broken Appium.

On Jul 5, 2018, at 3:08 PM, Alexei Barantsev <notifications@github.com<mailto:notifications@github.com>> wrote:


I should note that I couldn't reproduce the issue using the example project provided in the original report, I ran 100 iterations and got no failure. Win7, IE11, Selenium + IEDriverServer v.3.13.0

**Comment by barancev** [comment]
@RoakyWood Please answer my question instead of escalating the problem. Insisting on the mantra "the things are broken" does not help us figure out the root cause and fix it. Have you got a system error dialogue or just an exception in the client library? 

(My note was primarily directed to @jimevans to make him know that click is not globally broken on WIndows 7)

**Comment by RoakyWood** [comment]
I was merely responding to the ‘It works on my machine mantra’ which is equally unhelpful. I will look tomorrow for any system errors.

On Jul 5, 2018, at 6:45 PM, Alexei Barantsev <notifications@github.com<mailto:notifications@github.com>> wrote:


@RoakyWood<https://github.com/RoakyWood> Please answer my question instead of escalating the problem. Insisting on the mantra "the things are broken" does not help us figure out the root cause and fix it. Have you got a system error dialogue or just an exception in the client library?

(My note was primarily directed to @jimevans<https://github.com/jimevans> to make him know that click is not globally broken on WIndows 7)

**Comment by jimevans** [comment]
@RoakyWood “Seems to have broken Appium?” Sigh. At the risk of derailing this issue discussion, I don’t know what this comment means either, except that a potential break in Appium is entirely possible, as mentioned prominently in the [.NET bindings CHANGELOG](https://github.com/SeleniumHQ/selenium/blob/master/dotnet/CHANGELOG).

This is rapidly becoming counterproductive. I’m actually on holiday right now, responding to issue comments to the distress of my family. Surely, one can see that there’s no way for me to fix what I can’t reproduce, right?

**Comment by BrandonDudek** [comment]
@jimevans, enjoy your vacation!
For those of us having this issue, 3.12 will work for now.

We/I do realize that it is extremely difficult to fix a bug when you cannot reproduce it. I wish that I could be more help, but right now, I do not have access to a win 10 machine. I ran it on a Windows Server 2012 R2 machine and it had no errors. (The win 2012 server had IE 11.0.9600.19036.)

It could have something to do with Windows 7. Or maybe some IE configuration. (Or a combination.)

**Comment by RoakyWood** [comment]
Brandon,

I was able to reproduce it on Win 10 Enterprise Running IE 11 per my previous post. There was no error on Chrome or Edge on Win 10. I am going to check event logs to see if I can get more data.

RW

On Jul 6, 2018, at 10:21 AM, BrandonDudek <notifications@github.com<mailto:notifications@github.com>> wrote:


@jimevans<https://github.com/jimevans>, enjoy your vacation!
For those of us having this issue, 3.12 will work for now.

We/I do realize that it is extremely difficult to fix a bug when you cannot reproduce it. I wish that I could be more help, but right now, I do not have access to a win 10 machine. I ran it on a Windows Server 2012 R2 machine and it had no errors. (The win 2012 server had IE 11.0.9600.19036.)

It could have something to do with Windows 7. Or maybe some IE configuration. (Or a combination.)

**Comment by RoakyWood** [comment]
**Win 7 Event trace:**
Faulting application name: IEDriverServer.exe, version: 3.13.0.0, time stamp: 0x5b310d59
Faulting module name: msvcrt.dll, version: 7.0.7601.17744, time stamp: 0x4eeaf722
Exception code: 0xc0000005
Fault offset: 0x000099e4
Faulting process id: 0x406c
Faulting application start time: 0x01d4153fcf537091
Faulting application path: D:\Selenium\IEDriverServer.exe
Faulting module path: C:\Windows\syswow64\msvcrt.dll
Report Id: 16751527-8133-11e8-915e-64006a746b6b

**WIN 10 EVENT LOG**
Faulting application name: IEDriverServer.exe, version: 3.13.0.0, time stamp: 0x5b310d59
Faulting module name: ucrtbase.dll, version: 10.0.17134.1, time stamp: 0xbb0923b6
Exception code: 0xc0000005
Fault offset: 0x0005aca0
Faulting process id: 0x2ff8
Faulting application start time: 0x01d41541261f51a4
Faulting application path: C:\Selenium\IEDriverServer.exe
Faulting module path: C:\WINDOWS\System32\ucrtbase.dll
Report Id: 3f4525a1-e386-46e3-b09a-bfbc40f61bc7
Faulting package full name: 
Faulting package-relative application ID: 

VS 2017 ERROR OUTPUT
Test Name:	TestMethod
Test FullName:	Test.Test.TestMethod
Test Source:	C:\Users\blah\blah\blah
Test Outcome:	Failed
Test Duration:	0:00:36.884576

Result StackTrace:	
at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
 --- End of inner exception stack trace ---
    at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Net.PooledStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Net.Connection.SyncRead(HttpWebRequest request, Boolean userRetrievedStream, Boolean probeRead)
 --- End of inner exception stack trace ---
    at System.Net.HttpWebRequest.GetResponse()
   at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
 --- End of inner exception stack trace ---
    at OpenQA.Selenium.Remote.HttpCommandExecutor.MakeHttpRequest(HttpRequestInfo requestInfo)
   at OpenQA.Selenium.Remote.HttpCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.DriverServiceCommandExecutor.Execute(Command commandToExecute)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebElement.Execute(String commandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebElement.Click()
   at FMPopUpTest.FMPSCPopUp.FMPSCPopUpTestMethod() in C:\Users\BLAH

Result Message:	
Test method BLAH threw exception: 
OpenQA.Selenium.WebDriverException: A exception with a null response was thrown sending an HTTP request to the remote WebDriver server for URL http://localhost:2662/session/0eeef2a1-c013-4aa8-999c-acc9ad6fa47d/element/9ac5176c-6e21-4e94-a91d-23c0072e5374/click. The status of the exception was ReceiveFailure, and the message was: The underlying connection was closed: An unexpected error occurred on a receive. ---> System.Net.WebException: The underlying connection was closed: An unexpected error occurred on a receive. ---> System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host. ---> System.Net.Sockets.SocketException: An existing connection was forcibly closed by the remote host

**Comment by RoakyWood** [comment]
I have no urgency here either, I'm just running against Chrome in the meantime.

**Comment by BrandonDudek** [comment]
**I have managed to get rid of the error!**

I went to Windows Features, turned IE off, restarted, turned IE on, restarted, and now I am no longer getting any errors.

@RoakyWood, can you see if that works for you?

**Comment by RoakyWood** [comment]
IWebElement Click Action

Driver is IEDriverServer 32-bit
1. Clear Cache on IE
2. Close IE
3. Restart Win 7
4. Run Code
Result 1: Same error

- Change IEDriverServer to 64bit
- Run Code
Result 2: Success

Isolated error to Browser: IE 11 
Driver:                               IEDriverServer 32-bit 3.13
OS:                                    Win 7 SP 1 and/or Win 10
IDE:                                   VS Enterprise 2015 Version 14.0.25431.01 Update 3
.Net Framework:               Version 4.6.01590

**Comment by BrandonDudek** [comment]
@RoakyWood, did you try uninstalling and reinstalling IE?

**Comment by RoakyWood** [comment]
OK, here is a run down:

**SCENARIO I: Click() using IWeb Element**
Win 7, SP 1 Enterprise SeleniumStandalone 3.13 32 bit IEDriverServer 3.13.1

- Chrome PASS
- IE 11 FAIL

Uninstall IE 11 (NEVER do this)
Re-install IE 11 (HORRIBLE)

- IE 11 FAIL

**SCENARIO II: Click() using IWeb Element** 
Win 7, SP 1 Enterprise SeleniumStandalone 3.13 64 bit IEDriverServer 3.13.1

- Chrome PASS
- IE 11 PASS

**SCENARIO III: Click() using Actions/MoveTo**
Win 7, SP 1 Enterprise SeleniumStandalone 3.13 32 bit IEDriverServer 3.13.1

- Chrome PASS
- IE 11 PASS

**SCENARIO IV: Click() using IWeb Element**
Win 10 SeleniumStandalone 3.13 32 bit IEDriverServer 3.13.1

- Edge PASS
- IE 11 FAIL

**Data:**
For the IWebElement Click
Win 7 Driver-64 Bit: IE 11 Good, Chrome Good
Win 7 Driver-32 Bit: IE 11 Bad, Chrome Good
Win 10 Driver-32 Bit: IE 11 Bad, Edge Good

For the Actions/MoveTo
Win 7 Driver-32 Bit: IE 11 Good, Chrome Good

**Analysis:**
The problem is isolated to 32 Bit IEDriverServer using IWebElement Click Method, generates a browser-crashing  error on Win 7 or Win 10

**_Light thickens, and the crow Makes wing to th' roaky wood._**

**Comment by jimevans** [comment]
@RoakyWood Can you please retry this repro with the 32-bit `IEDriverServer.exe` version 3.13.0.3 [checked into the source tree](https://github.com/SeleniumHQ/selenium/blob/master/cpp/prebuilt/Win32/Release/IEDriverServer.exe)?

**Comment by RoakyWood** [comment]
Supergreen! Looks good, nice work!
RW

**Comment by jimevans** [comment]
@BrandonDudek Your turn. Does the fix provided in 3.13.0.3 resolve your crash too?

**Comment by jimevans** [comment]
Closing this as fixed by 94bc66e. If other users are still experiencing crashes, please open a new issue.

**Comment by BrandonDudek** [comment]
Finally tested version 3.13.0.5. It worked for me.
Thanks! :)
:+1:

**Comment by guama** [comment]
I was getting the error with 64 bit 3.13 and saw someone said to use an earlier version so I went back down to 3.12 and got this error but I think I can fix that. 
selenium.common.exceptions.SessionNotCreatedException: Message: Unexpected error launching Internet Explorer. Protected Mode settings are not the same for all zones. Enable Protected Mode must be set to the same value (enabled or disabled) for all zones.

Thank you for your help it is appreciated.

**Comment by jimevans** [comment]
@guama And using the most recently released 3.14?

**Comment by sksajjan** [comment]
IEDriverServer.exe version 3.14.0 is not having issue with closing browser driver, however, we are seeing an issue with sendKeys for textArea fields. sendKeys works fine for the input tags but not for textarea
